src/datasources/notifications/notifications.datasource.ts

```diff
@@ -59,21 +59,32 @@ export class NotificationsDatasource implements INotificationsDatasource {
       await Promise.all(
         args.upsertSubscriptionsDto.safes.map(async (safe) => {
           try {
-            // 1. Upsert subscription
+            // 1. Delete previous subscriptions (and by cascade, notification types)
+            //    to avoid duplicates, and clearing "unknown" subscriptions
+            await sql`
+              DELETE FROM notification_subscriptions
+              WHERE chain_id = ${safe.chainId}
+              AND safe_address = ${safe.address}
+              AND (
+                signer_address = ${args.signerAddress ?? null}
+                OR (
+                  signer_address IS NULL
+                )
+              )
+              AND push_notification_device_id = ${device.id}
+            `;
```

Comment:
The subscription is still device based so this shouldn't cause issues of "unsubscriptions".